Unix/http/httpauth.c

```diff
@@ -1639,19 +1711,52 @@ MI_Boolean IsClientAuthorized(_In_ Http_SR_SocketData * handler)
     if (Strncasecmp(headers->authorization, AUTHENTICATION_BASIC, AUTHENTICATION_BASIC_LENGTH) == 0)
     {
         handler->httpAuthType = AUTH_METHOD_BASIC;
-        if (!headers->username || !headers->password || 0 != AuthenticateUser(headers->username, headers->password))
+        if (!headers->username || !headers->password)
         {
             handler->httpErrorCode = HTTP_ERROR_CODE_UNAUTHORIZED;
             auth_response = (char *)RESPONSE_HEADER_UNAUTH_FMT;
             response_len  = RESPONSE_HEADER_UNAUTH_FMT_LEN;
 
-            trace_HTTP_UserAuthFailed("user not authenticated");
+            trace_HTTP_UserAuthFailed("missing user name or password");
             handler->authFailed = TRUE;
 
             _SendAuthResponse(handler, auth_response, response_len);
             return FALSE;
         }
 
+        if (0 == IsRoot())
+        {
+            if (0 != AuthenticateUser(headers->username, headers->password))
+            {
+                handler->httpErrorCode = HTTP_ERROR_CODE_UNAUTHORIZED;
+                auth_response = (char *)RESPONSE_HEADER_UNAUTH_FMT;
+                response_len  = RESPONSE_HEADER_UNAUTH_FMT_LEN;
+
+                trace_HTTP_UserAuthFailed("user not authenticated");
+                handler->authFailed = TRUE;
+                
+                _SendAuthResponse(handler, auth_response, response_len);
```

Comment:
Not sure which failure path you are referring to.  These blocks of code deal with failure cases.  If you mean _SendAuthResponse...it's a void function, but does log if it fails.